src/Components/Availability/useAvailability.ts

```diff
@@ -0,0 +1,112 @@
+import { useCreation, useRequest } from "ahooks";
+import { useEffect, useState } from "react";
+import { BehaviorSubject } from "rxjs";
+import { Station } from "~/Helpers/Entities";
+import { Logger } from "~/Helpers/Logger";
+import { useStatus } from "~/Services/Status";
+import { Models } from "~/Services/Status.Models";
+
+const log = new Logger("useAvailability");
+
+interface ServiceAvaEntity {
+  id: number
+  name: string
+  availability: AvaliaEntity[]
+  region: string
+}
+
+interface AvaliaEntity {
+  year: number
+  month: number
+  percentage: number
+}
+
+interface IAvailability {
+  RS: Models.IRegionService,
+  Percentages: number[]
+}
+
+/**
+ * Custom hook to manage availability data based on the provided category and topic.
+ * This hook interacts with the status service and fetches availability data from a backend service.
+ * It processes and filters the data to return availability information for a specific region and category.
+ *
+ * @param category - The category of the service for which availability data is required.
+ * @param topic - The topic used to subscribe to region updates.
+ * @returns An array of availability information objects.
+ *
+ * @remarks
+ * This hook uses various state and effect hooks to manage and update the availability data.
+ * It also utilizes a custom logger for debugging purposes.
+ * The data fetching is performed using a request hook with caching enabled.
+ *
+ * @author Aloento
+ * @since 1.0.0
+ * @version 0.1.0
+ */
+export function useAvailability(category: Models.ICategory, topic: string) {
+  const { DB } = useStatus();
+
+  const [region, setRegion] = useState(DB.Regions[0]);
+  const regionSub = useCreation(
+    () => Station.get<BehaviorSubject<Models.IRegion>>(topic), []);
+
+  useEffect(() => {
+    const sub = regionSub.subscribe(setRegion);
+    return () => sub.unsubscribe();
+  }, []);
+
+  const [avas, setAvas] = useState<IAvailability[]>(
+    () => DB.RegionService
+      .filter(x => x.Region.Id === region.Id)
+      .filter(x => x.Service.Category.Id === category.Id)
+      .map(x => ({
+        RS: x,
+        Percentages: Array(6).fill(100)
+      })));
+
+  const url = process.env.SD_BACKEND_URL;
```

Comment:
maybe better to extract this type of const from method?

Reply:
After compiling, it will be statically replaced with a const string and inline.
const url = "https://...";
so, it's fine here, XD